src/odemis/dataio/tiff.py

```diff
@@ -1854,83 +1862,110 @@ def _saveAsMultiTiffLT(filename, ldata, thumbnail, compressed=True, multiple_fil
+
+    for fifd, das in sorted_groups:
+        if len(das) == 0:
+            continue  # Something is wrong
+        elif len(das) == 1:
+            data = das[0]
+            # Just normal output
+            tags = _convertToTiffTag(data.metadata)
+            # if metadata indicates YXC format just handle it as RGB
+            if data.metadata.get(model.MD_DIMS) == 'YXC' and data.shape[-1] in (3, 4):
+                write_rgb = True
+                hdim = data.shape[:-3]
+            # TODO: handle RGB for C at any position before and after XY, but iif TZ=11
+            # for data > 2D: write as a sequence of 2D images or RGB images
+            elif data.ndim == 5 and data.shape[0] == 3:  # RGB
+                # Write an RGB image, instead of 3 images along C
+                write_rgb = True
+                hdim = data.shape[1:3]
+                # CTZYX -> TZCYX
+                data = numpy.moveaxis(data, 0, 2)  # move C axis near YX
+            elif data.ndim == 5:
+                write_rgb = False
+                # CTZYX -> TZCYX  (for ImageJ compatible ordering)
+                data = numpy.moveaxis(data, 0, 2)
+                hdim = data.shape[:-2]
+            else:  # YX
+                write_rgb = False
+                hdim = data.shape[:-2]
+
+            for i in numpy.ndindex(*hdim):
+                # Save metadata (before the image)
+                for key, val in tags.items():
+                    try:
+                        f.SetField(key, val)
+                    except Exception:
+                        logging.exception("Failed to store tag %s with value '%s'", key, val)
+                if data.dtype in [numpy.int64, numpy.uint64]:
+                    c = None  # libtiff doesn't support compression on these types
+                else:
+                    c = compression
+                write_image(f, data[i], write_rgb=write_rgb, compression=c, pyramid=pyramid)
+
         else:
-            write_rgb = False
-            hdim = data.shape[:-2]
+            # len(das) > 1 => list of DataArrays to represent the C dimension
+            # the shape of all das are same: 1TZYX, or ZYX, or TZYX
+            # len(das) is to be used as C
 
-        for i in numpy.ndindex(*hdim):
-            # Save metadata (before the image)
-            for key, val in tags.items():
-                try:
-                    f.SetField(key, val)
-                except Exception:
-                    logging.exception("Failed to store tag %s with value '%s'", key, val)
-            if data[i].dtype in [numpy.int64, numpy.uint64]:
-                c = None # libtiff doesn't support compression on these types
-            else:
-                c = compression
-            write_image(f, data[i], write_rgb=write_rgb, compression=c, pyramid=pyramid)
+            write_rgb = False
+            hdim = das[0].shape[:-2]  # 1TZ or TZ or Z
```

Comment:
Do the tests cover all these possible dimensions?

Reply:
I will add the tests